### PR TITLE
Show failed dlopen result in verbose mode

### PR DESCRIPTION
**Pull request by ianshmean**
Hoping to help debug dlopen issues. Should be helpful during a CI sessions

cc. @staticfloat @giordano

### CODE REVIEW

File: src/Products.jl [hunk 1]
```diff
@@ -178,6 +178,8 @@ function locate(lp::LibraryProduct; verbose::Bool = false,
 
                     if verbose
                         @info("$(dl_path) cannot be dlopen'ed")
```
Comment (staticfloat) [inline src/Products.jl]: Let's have `res` be a second parameter to the above `@info` call.

**Comment by staticfloat** [comment]
Ah, yeah, we don't have nice things on 1.0.  :(

**Comment by staticfloat** [comment]
Note that because this isn't using "isolated" mode, some things may fail the first time then succeed the second time.  But that's still useful information for us!

File: src/Products.jl (hunk 1 above)
Comment (ianshmean) [reply src/Products.jl]: done, with new try catch

**Pull request merged**